src/main/scala/higherkindness/rules_scala/workers/zinc/compile/ZincRunner.scala

```diff
@@ -212,17 +212,25 @@ object ZincRunner extends WorkerMain[ZincRunnerWorkerConfig] {
         isWorker,
       )
 
+    val shouldIncludeSourceRoot = !scalaInstance.actualVersion.startsWith("0.") &&
+      scalaInstance.actualVersion.startsWith("3")
+
+    val scalacOptions =
+      workRequest.plugins.view.map(p => s"-Xplugin:$p").toArray ++
+        workRequest.compilerOptions ++
+        workRequest.compilerOptionsReferencingPaths.toArray ++
+        (if (shouldIncludeSourceRoot)
```

Comment:
Could we write a test to ensure that the TASTy files are deterministically generated, regardless of the work directory? It's difficult to control which work directory Bazel uses under multiplex sandboxing, so perhaps we could just build the same target a few times.